Add Go 1.13

rvolosatovs opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Upstream update
Seems like progress on #68101 stalled for now, meanwhile I already have a working Go 1.13 derivation, which I have been successfully using for ~1.5 working days~ ~2 weeks~ a while.

I did not use a function for skipping tests like @adisbladis suggested and implemented in https://github.com/adisbladis/nixpkgs/tree/go_1_13, IMO this is out of scope of this PR - consistency is more important here.
I think stylistic/organizational changes belong in #35381

~Packages still broken(there might be more):~
- ~deepin~

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @cstrahan @orivej @velovix @Mic92 @kalbasit

rvolosatovs committed:
go_1_13: init at 1.13.1

rvolosatovs committed:
go: default to go_1_13

rvolosatovs commented:
@GrahamcOfBorg build go direnv

adisbladis commented:
@rvolosatovs I did a `nox-review` on my branch and found the following failures:
```
 /nix/store/0xaddkf0ma3h0mlny3273icn2g3ixzgk-lnd-0.7.1-beta.drv
  /nix/store/137jp3qyvllfy08z6pvnsxlr6f0w37zg-magnetico-0.8.1.drv
  /nix/store/w16anbv6bcjsfi2jmq3qsf8l4shki2rh-qtwebkit-5.212-alpha-01-26-2018.drv
  /nix/store/ll4nzrllg4hijlpyysp2zx3639607s78-phantomjs-2.1.1.drv
  /nix/store/2dnvggx9qzmbdq147nf8vfy59akd2rk4-grafana-6.3.5.drv
  /nix/store/39xbs03k9qpbxlz4p6g332q69j2r182w-grpcui-0.2.0.drv
  /nix/store/5550hrgpg9xaf7rxz5wx07f5s6rdxpdf-conftest-0.7.0.drv
  /nix/store/5ip41wik25fxsbrha7y54v2h4h6x69c5-vault-1.2.2-go-modules.drv
  /nix/store/6bgchfqk6wh94w7g698kiyiy44r12zdk-lazydocker-0.7.drv
  /nix/store/6d3xh04ma6vykabwphfx26hgfqf4hzim-cue-0.0.3.drv
  /nix/store/6qa5yf2bw16zs0dyjrq35b0fv5jb8l8m-helmfile-0.79.3.drv
  /nix/store/8g4bjj61lz3xy6y2kyhwhi1mmd9xrxsc-bazel-watcher-0.10.3.drv
  /nix/store/8m67bqhh8z4qh71xqf7y2knc84k0dfpc-ctop-0.7.2.drv
  /nix/store/vqyw3sl6al0346kvq7zdhd1p6xxcgv3q-terraform-provider-elasticsearch-0.7.0.drv
  /nix/store/zwqh87wy5aa8xz9x100928qzibw14s7p-tf-plugin-env.drv
  /nix/store/v1zb0437a90y72fmlv3jz8klsf9yxj30-terraform-0.11.14.drv
  /nix/store/bhl5ik1dva2ad2bjvjrqfriyf7dnhfiv-terragrunt-0.17.4.drv
  /nix/store/cnpl55ripjq788bzafn6d76kkh426adq-ghq-0.12.6.drv
  /nix/store/hgga50qy7qmji6ifqp9qwxv82q3wy84q-terraform-lsp-0.0.5.drv
  /nix/store/hmvq28ga6agsignci3mdm9nhvjbnbkwh-prometheus_varnish_exporter-1.5.1.drv
  /nix/store/inkjfbyzkhcb51z4xf5i82k93a2hrh8f-scaleway-cli-1.17.drv
  /nix/store/k2z4p92p9vma8i0d589ws1qx4qvkqpa5-gotestsum-0.3.5.drv
  /nix/store/khfagkcdkygmsjj782zp6s6iw17jw228-curlie-1.2.0.drv
  /nix/store/kngxad53xhiy0cbf2y8aap6y86hdjxr4-prow-unstable-2019-08-14.drv
  /nix/store/lp1a7y4byli5l4zgdsnfbp52npd1np4c-packr-2.5.2.drv
  /nix/store/mhykzrg6r4x2alv8rppngzkqcwxaw417-tendermint-0.32.2.drv
  /nix/store/mq3w6khxlrqpx4yqg4697bz754qla9jg-restic-0.9.5.drv
  /nix/store/ng8f7mizfqpmqzb4phx3qmlvf3zrnq0q-cri-o-1.15.0.drv
  /nix/store/q54c9h6smxg2sxka6ib9qdx4zjddx9fg-openshift-origin-3.11.0.drv
  /nix/store/q57x1sg297v01dw7mk51kmq7ylqmn0ri-wtf-0.21.0.drv
  /nix/store/s4mlrqhdd26f8429chwxhsby652nk76x-git-lfs-2.7.2.drv
  /nix/store/sjmc0hllxzpa6lhalac6wp1y8zfs2l2q-vault-1.2.2.drv
  /nix/store/spf0xv7l7pf18r1ayv2rdngal29c6vrn-cockroach-19.1.1.drv
  /nix/store/x54vx64xkcfrlg46yqp0h82nw8xqq6jc-consul-1.6.0.drv
  /nix/store/x5pfsam5chjzrnx38969sxfpiar4522l-ipfs-0.4.21.drv
  /nix/store/xl2iyn7cznni9ri5zq65v2ydsd9lcfwp-blockbook-0.3.1.drv
  /nix/store/y9bcmy7ycwvr3wnqmdv68fnawv5d4bd5-deepin-wm-1.9.38.drv
  /nix/store/ydbmnf9372y75pa2r94zdmvwq3z99pl8-minishift-1.34.0.drv
  /nix/store/z8jw1w8i61ibggxbj6c772jd6r04s2gz-skydive-0.17.0.drv
  /nix/store/zjv7saninzagsnw6g41jqjg7w95gcys9-eksctl-0.4.3.drv
```

rvolosatovs commented:
Indeed, I see. At least in case of `ghq`, the package is incompatible with Go 1.13, due to a dependency on `xerrors`, which in turn is incompatible with Go 1.13
I opened an issue and PR upstream https://github.com/motemen/ghq/issues/192 https://github.com/motemen/ghq/pull/193
I assume we can mitigate most of the failures by just using the go 1.12 for the builds

rvolosatovs commented on pkgs/applications/version-management/git-and-tools/ghq/default.nix:
```diff
@@ -1,29 +1,31 @@
 
   postInstall = ''
-    install -m 444 -D ${src}/zsh/_ghq $bin/share/zsh/site-functions/_ghq
+    install -m 444 -D ${src}/zsh/_ghq $out/share/zsh/site-functions/_ghq
```
Shall we add a `bin` output here for backwards compatibility?
Putting this in `bin` seems very wrong to me

Mic92 replied:
Due to historical reasons we have `$bin` in `buildGoPackage` as `$out` contained the source code of it. At some point we got rid of the source code but kept `$bin` for compatibility reasons. We corrected this when introducing `buildGoModule`.

rvolosatovs commented:
@GrahamcOfBorg build ghq cockroach ipfs

rvolosatovs commented:
@GrahamcOfBorg build ipfs

kalbasit commented on pkgs/development/compilers/go/1.13.nix:
```diff
@@ -0,0 +1,239 @@
+    homepage = http://golang.org/;
+    description = "The Go Programming language";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ cstrahan orivej velovix mic92 rvolosatovs ];
```
```suggestion
    maintainers = with maintainers; [ cstrahan orivej velovix mic92 rvolosatovs kalbasit ];
```

I missed the preview PRs, do you mind adding me here so moving forward I get review request?

kalbasit replied:
@rvolosatovs I got removed again, probably due to a force-push.

Mic92 commented on pkgs/applications/version-management/git-and-tools/ghq/default.nix:
```diff
@@ -1,29 +1,31 @@
   '';
 
+  patches = [
+    ./update-xerrors-dep.patch # remove once https://github.com/motemen/ghq/pull/193 is merged
```
```
(fetchpatch { 
  url = "https://github.com/motemen/ghq/commit/38ac89e60e60182b5870108f9753c9fe8d00e4a6.patch";
  sha256 = "....";
})
```

rvolosatovs replied:
https://github.com/NixOS/nixpkgs/pull/68160

rvolosatovs commented:
@adisbladis do you have fixes for (any of) packages broken by 1.13 update?
If so, maybe you can open PRs for them or I can cherry-pick them here?

rvolosatovs committed:
all-packages: use go 1.12 for consul

rvolosatovs commented:
@GrahamcOfBorg build consul ipfs ghq

rvolosatovs committed:
all-packages: use go 1.12 for perkeep

rvolosatovs commented:
Fixed `conftest` in https://github.com/NixOS/nixpkgs/pull/68438

rvolosatovs commented:
@GrahamcOfBorg build conftest perkeep

rvolosatovs committed:
ctop: ensure go 1.13 compatibility

rvolosatovs commented:
@GrahamcOfBorg build ctop conftest

rvolosatovs commented:
Fixed `cue` in #68451

rvolosatovs commented:
Fixed `git-lfs` in #68455

rvolosatovs commented:
These packages are still broken by the update(there are probably more):
- bazel-watcher
- cri-o
- curlie
- deepin
- gotestsum
- grpcui
- helmfile
- lnd
- packr
- prow
- restic
- scaleway-cli
- tendermint
- terraform-lsp
- terraform-provider-elasticsearch
- wtf

rvolosatovs commented:
Fixed `lazydocker` in https://github.com/NixOS/nixpkgs/pull/68460

rvolosatovs committed:
all-packages: use go 1.12 for wtf

rvolosatovs committed:
all-packages: use go 1.12 for bazel-watcher

rvolosatovs committed:
all-packages: use go 1.12 for cri-o

rvolosatovs committed:
all-packages: use go 1.12 for curlie

rvolosatovs committed:
all-packages: use go 1.12 for gotestsum

rvolosatovs committed:
all-packages: use go 1.12 for grpcui

rvolosatovs commented:
Fixed `helmfile` in https://github.com/NixOS/nixpkgs/pull/68943

rvolosatovs committed:
all-packages: use go 1.12 for lnd

rvolosatovs committed:
all-packages: use go 1.12 for scaleway-cli

rvolosatovs committed:
all-packages: use go 1.12 for tendermint

rvolosatovs commented:
`buildGo113Module` consistently generates a different `modSha256` than `buildGo112Module`. @kalbasit any idea what might be going wrong?

rvolosatovs committed:
all-packages: use go 1.12 for restic

rvolosatovs committed:
all-packages: use go 1.12 for packr

rvolosatovs committed:
all-packages: use go 1.12 for terraform-lsp

rvolosatovs committed:
all-packages: use go 1.12 for terraform-providers.elasticsearch

rvolosatovs committed:
all-packages: use go 1.12 for prow

c00w commented:
I'm running a local nix-review pr 68135 and seeing the following failures (will update as it finished).

[/nix/store/c4arazdqach2nama5q2vzi15pbmc03vv-conftest-0.12.0.drv](https://github.com/NixOS/nixpkgs/files/3623860/failure.txt)

c00w committed:
all-packages: use go 1.12 for grafana

c00w committed:
all-packages: use go 1.12 for magnetico

Co-authored-by: Roman Volosatovs <rvolosatovs@riseup.net>

c00w committed:
all-packages: use go 1.12 for openshift

c00w committed:
all-packages: use go 1.12 for kubernetes

rvolosatovs commented:
Thanks for looking into this!
`conftest` should have been fixed by https://github.com/NixOS/nixpkgs/pull/68438 :thinking:

c00w committed:
all-packages: use go 1.12 for cockroachdb

rvolosatovs commented:
@c00w I reworded your commits, split one, added a comment and added them to the PR. hope you don't mind

c00w committed:
conftest: 0.12.0 -> 0.14.0, disable go sum db.

The go sum db is a hosted service that appears to be generating a new
inclusion proof each time this build is run (rather than using the one
from the previous run). Since this ends up the module directory + thus
hash, it's breaking deterministic builds.

Co-authored-by: Roman Volosatovs <rvolosatovs@riseup.net>

rvolosatovs commented:
I'm afraid the module hash computation with go 1.13 is non-deterministic.
E.g. consider `conftest`:
Intially I got `0l44840x0dxa4j8ghzdjbpf1ckf9mssaxx4r3mb9y1d34pksdmnf` (https://github.com/NixOS/nixpkgs/commit/186e57525771e251e6389304a05c2a09171d47c4#diff-350e2f990b84dd24d5f9316a05eb9063R11) `modSha256` and successfully built it with `go 1.13`

Then after https://github.com/NixOS/nixpkgs/pull/68438#issuecomment-530458347 (I assume @Mic92 was using go 1.12?) I changed the hash and PR got merged.

Now the specified `modSha256` corresponds to a different hash than `go 1.13` expects, hence https://github.com/NixOS/nixpkgs/pull/68135/commits/f8338933138801bbffe58439d8bcef61357c2855. On `go 1.13` trying to build the package with the modules from `go 1.12`(the cached modules) result in:
```
build github.com/instrumenta/conftest/cmd: cannot load github.com/googleapis/gax-go/v2: github.com/googleapis/gax-go@v2.0.0+incompatible: reading file:///nix/store/4ykyhkif3vaxfw66bmc4mgxvvm24p82s-conftest-0.12.0-go-modules/github.com/googleapis/gax-go/@v/v2.0.0+incompatible.zip: Not Found
```

So now regenerating the module hash on `go 1.13` I get different hashes every time I try to regenerate it:
- `1kv5lhxpgacb7fv4zsx0j32ksv0v80xj6iabr5hps44zb69dn31s`
- `023j34yjgp3asysrjbls86nvhf51d1n86ll3db493rsivlcm0m5q`
- `0v4nl1p346kp0byq1g5xdlphpra2cp08r4ainzq892dj0g0dv09r`

Any idea what could be causing the non-determinism?
cc @kalbasit @Mic92

c00w commented:
@rvolosatovs - That's exactly why I made them :D

kalbasit commented:
@rvolosatovs one way to figure out what's causing the inconsistency is to compare the versions go-modules in the nix store. For instance, run `nix-build -A conftest.go-modules` to get the nix store path, then try a second time to get a second nix store path and finally `diff -Naur /nix/store/1... /nix/store/2...`.

We can discuss the solution after we figure out what's different.

c00w commented:
FYI - packetbeat can be ignored since it's broken in master and is a 6 month old outstanding issue?
https://github.com/elastic/beats/issues/11054

c00w committed:
all-packages: use go 1.12 for prometheus and prometheus-varnish-exporter

These both fail to work with go1.13 due to test issues, downgrading to
1.12

c00w commented:
wal-g can be ignored since it's broken in master.

c00w commented:
vault is also broken in master

c00w commented:
I have the conftest diff - looks like somehow we're getting some random data that's inconsistent? I'm definitely not clear about where the tile data is coming from.

[conftest-diff.log](https://github.com/NixOS/nixpkgs/files/3634191/conftest-diff.log)

c00w commented:
I've downgraded prometheus and conftest to go 1.12 at https://github.com/c00w/nixpkgs/tree/go1_13_2
At this point I believe all remaining failures (packetbeat, wal-g, and vault) are already broken in master for x64. I'll try to get a run off on aarch64, but I'll need to actually get my raspberry pi build cluster set up.

rvolosatovs commented:
It seems like Go records the lookup/download time? Maybe we can mitigate that by making Go think we're at Unix epoch in `buildGoModule`?

@c00w why downgrade `conftest` if it builds fine on go 1.13? Let's fix the non-determinism instead.

Mic92 commented:
It looks like their sumdb is causing issues: `go/pkg/mod/cache/download/sumdb/sum.golang.org`

They uses that to check for package hashes. Maybe we can disable/delete it before calculating the hash sum?

c00w commented:
To disable it we need to set the environment variable
GONOSUMDB=*
See https://golang.org/cmd/go/#hdr-Module_configuration_for_non_public_modules
This is probably safe since we are encoding the hash already. I think this may be something weird with conftest since they do a bunch of overrides and I suspect one of their downloads somehow includes some constantly changing files (the tile directory).

Mic92 commented:
@c00w The feature itself looks quite useful as it enhances security when fetching go modules for the first time. If there is an easy way to still use this database, but delete the artifacts before calculating the hash sum, I would prefer that.

kalbasit commented:
> It seems like Go records the lookup/download time? Maybe we can mitigate that by making Go think we're at Unix epoch in `buildGoModule`?

Can you paste in the diff and/or upload the nix store locations you got?

rvolosatovs commented:
@c00w already did that, see https://github.com/NixOS/nixpkgs/pull/68135#issuecomment-533401277

marsam committed:
documize-community: disable go module build

marsam commented:
I rebased on master and fixed documize-community  https://github.com/NixOS/nixpkgs/compare/master...marsam:pr/68135
do you mind if I push my changes to this PR? or you prefer cherry picking them?

rvolosatovs commented:
@marsam I rebased on latest `master` and cherry-picked https://github.com/NixOS/nixpkgs/commit/d23a51b4921cdf19e48c0637974ae71b051c5841
Please add commits on top of this PR if required, but I would prefer if you don't force-push.

marsam commented:
nix-review on darwin shows as failing:
```
17 package failed to build:
circleci-cli conftest dcrd dcrwallet git-lfs gitaly hyperledger-fabric openshift packetbeat prometheus prometheus-apcupsd-exporter prometheus-blackbox-exporter prometheus-varnish-exporter qsyncthingtray wal-g wireguard-go wireguard-tools
```
later I'm going to review each one
<details>
<summary>
nix-review on darwin output
</summary>

```
$ nix-review pr 68135
$ git fetch --force https://github.com/NixOS/nixpkgs master:refs/nix-review/0 pull/68135/head:refs/nix-review/1
$ git worktree add /Users/marsam/.cache/nix-review/pr-68135-9/nixpkgs 2202601623faf37a7ed31dfedf4c78ed3931cc7a
Preparing worktree (detached HEAD 2202601623f)
Updating files: 100% (19883/19883), done.
HEAD is now at 2202601623f perlPackages.MustacheSimple: init at 1.3.6
$ git merge --no-commit d7225664a2714fb1b400a78841e432aaec6ac5ab
Auto-merging pkgs/top-level/all-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --max-jobs 8 --option build-use-sandbox true -f /Users/marsam/.cache/nix-review/pr-68135-9/build.nix
builder for '/nix/store/c115pjb5wwa37hyv069bd3fdxpyscbmf-circleci-cli-0.1.5879.drv' failed with exit code 1; last 10 log lines:
  go: finding github.com/mattn/go-isatty v0.0.4
  go: finding github.com/lunixbochs/vtclean v0.0.0-20170504063817-d14193dfc626
  go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
  go: finding github.com/tcnksm/go-gitconfig v0.1.2
  go: finding github.com/ulikunitz/xz v0.5.4
  go: finding github.com/google/go-github v15.0.0+incompatible
  go: finding golang.org/x/oauth2 v0.0.0-20180724155351-3d292e4d0cdc
  go: finding golang.org/x/net v0.0.0-20181220203305-927f97764cc3
  go: finding github.com/google/go-querystring v0.0.0-20170111101155-53e6ce116135
  build github.com/CircleCI-Public/circleci-cli: cannot load github.com/gobuffalo/packr/v2: github.com/gobuffalo/packr@v1.21.0: reading file:///nix/store/hh9hrfdg3gbqz37sa3hx44mkzkxpkkyw-circleci-cli-0.1.5879-go-modules/github.com/gobuffalo/packr/@v/v1.21.0.zip: Not Found
builder for '/nix/store/yyir3s7ck0n010kb0kk3453ac0cg9ln1-git-lfs-2.8.0.drv' failed with exit code 1; last 7 log lines:
  unpacking sources
  unpacking source archive /nix/store/si8k81lcp6snbj466g0jk5qf52x0hhc9-source
  source root is source
  patching sources
  configuring
  building
  go: github.com/alexbrainman/sspi@v0.0.0-20180125232955-4729b3d4d858: Get https://proxy.golang.org/github.com/alexbrainman/sspi/@v/v0.0.0-20180125232955-4729b3d4d858.mod: dial tcp: lookup proxy.golang.org: no such host
builder for '/nix/store/4gwn40xw7c5nmxg6p4wmvk9ypy0k7lm8-dcrd-1.1.2.drv' failed with exit code 1; last 10 log lines:
  unpacking source archive /nix/store/4r9ixi5q1w2y4bnnh76rs603n8r44dyk-dcrd
  source root is dcrd
  patching sources
  configuring
  building
  The following issues were found in Gopkg.toml:
  
    ✗ unable to deduce repository and source type for "golang.org/x/crypto": unable to read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://golang.org/x/crypto?go-get=1": Get http://golang.org/x/crypto?go-get=1: dial tcp: lookup golang.org: no such host
  
  ProjectRoot name validation failed
builder for '/nix/store/js9z6yf77r0lh6g0p2yzvbxxlbwjgfnd-dcrwallet-1.1.2.drv' failed with exit code 1; last 10 log lines:
  configuring
  building
  The following issues were found in Gopkg.toml:
  
    ✗ unable to deduce repository and source type for "golang.org/x/net": unable to read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://golang.org/x/net?go-get=1": Get http://golang.org/x/net?go-get=1: dial tcp: lookup golang.org: no such host
    ✗ unable to deduce repository and source type for "google.golang.org/grpc": unable to read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://google.golang.org/grpc?go-get=1": Get http://google.golang.org/grpc?go-get=1: dial tcp: lookup google.golang.org: no such host
    ✗ unable to deduce repository and source type for "golang.org/x/sync": unable to read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://golang.org/x/sync?go-get=1": Get http://golang.org/x/sync?go-get=1: dial tcp: lookup golang.org: no such host
    ✗ unable to deduce repository and source type for "golang.org/x/crypto": unable to read metadata: unable to fetch raw metadata: failed HTTP request to URL "http://golang.org/x/crypto?go-get=1": Get http://golang.org/x/crypto?go-get=1: dial tcp: lookup golang.org: no such host
  
  ProjectRoot name validation failed
builder for '/nix/store/3xsphpidvpsrl5nnmbhf43kvsj2z15as-apcupsd-exporter-unstable-2019-03-14.drv' failed with exit code 1; last 10 log lines:
  github.com/mdlayher/apcupsd_exporter.testCollector(0xc000144200, 0x15629a0, 0xc0001a4000, 0x0, 0x0, 0x0)
  	/private/tmp/nix-build-apcupsd-exporter-unstable-2019-03-14.drv-0/go/src/github.com/mdlayher/apcupsd_exporter/apcupsdexporter_test.go:18 +0x14f
  github.com/mdlayher/apcupsd_exporter.TestUPSCollector.func1(0xc000144200)
  	/private/tmp/nix-build-apcupsd-exporter-unstable-2019-03-14.drv-0/go/src/github.com/mdlayher/apcupsd_exporter/upscollector_test.go:72 +0x6e
  testing.tRunner(0xc000144200, 0xc00013a240)
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:909 +0xc9
  created by testing.(*T).Run
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:960 +0x350
  FAIL	github.com/mdlayher/apcupsd_exporter	0.019s
  FAIL
builder for '/nix/store/i81gc3r1i99yd9smkhxi3bj4f43i3sqy-blackbox_exporter-0.15.1.drv' failed with exit code 1; last 10 log lines:
  net/http/httptest.NewServer(0x17ae920, 0x1701f48, 0x0)
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/net/http/httptest/server.go:99 +0x27
  github.com/prometheus/blackbox_exporter.TestPrometheusTimeoutHTTP(0xc00012a300)
  	/private/tmp/nix-build-blackbox_exporter-0.15.1.drv-0/go/src/github.com/prometheus/blackbox_exporter/main_test.go:46 +0x4e
  testing.tRunner(0xc00012a300, 0x1701f58)
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:909 +0xc9
  created by testing.(*T).Run
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:960 +0x350
  FAIL	github.com/prometheus/blackbox_exporter	0.023s
  FAIL
builder for '/nix/store/lrjz5956nz8yd37plihwnpyhb54xyk5w-packetbeat-6.8.3.drv' failed with exit code 1; last 10 log lines:
  github.com/elastic/beats/vendor/github.com/ericchiang/k8s/apis/apps/v1beta1
  github.com/elastic/beats/vendor/github.com/ericchiang/k8s/apis/extensions/v1beta1
  github.com/elastic/beats/libbeat/common/kubernetes
  github.com/elastic/beats/libbeat/autodiscover/providers/kubernetes
  github.com/elastic/beats/libbeat/processors/add_kubernetes_metadata
  github.com/elastic/beats/packetbeat/processor/add_kubernetes_metadata
  github.com/elastic/beats/libbeat/cmd/instance
  github.com/elastic/beats/libbeat/cmd/test
  github.com/elastic/beats/libbeat/cmd/export
  github.com/elastic/beats/libbeat/cmd
builder for '/nix/store/8ch1ckdjvzk1ikdkmak2cmm3yh15zsj2-prometheus_varnish_exporter-1.5.1.drv' failed with exit code 1; last 10 log lines:
    -web.health-path string
      	Path under which to expose healthcheck. Disabled unless configured.
    -web.listen-address string
      	Address on which to expose metrics and web interface. (default ":9131")
    -web.telemetry-path string
      	Path under which to expose metrics. (default "/metrics")
    -with-go-metrics
      	Export go runtime and http handler metrics
  FAIL	github.com/jonnenauha/prometheus_varnish_exporter	0.015s
  FAIL
builder for '/nix/store/ijv3ffjb055ayhmlzgc81wy68rwkba4i-ruby2.6.4-grpc-1.19.0.drv' failed with exit code 1; last 10 log lines:
  	--with-make-prog
  	--without-make-prog
  	--srcdir=.
  	--curdir
  	--ruby=/nix/store/9lywmsbfjcwjgdfvp6p9whx8gl3hyf71-ruby-2.6.4/bin/$(RUBY_BASE_NAME)
  
  extconf failed, exit code 1
  
  Gem files will remain installed in /nix/store/fymzmgxbhyh3j32k0kbls1gflfl0jvjy-ruby2.6.4-grpc-1.19.0/lib/ruby/gems/2.6.0/gems/grpc-1.19.0 for inspection.
  Results logged to /nix/store/fymzmgxbhyh3j32k0kbls1gflfl0jvjy-ruby2.6.4-grpc-1.19.0/lib/ruby/gems/2.6.0/extensions/x86_64-darwin-17/2.6.0/grpc-1.19.0/gem_make.out
cannot build derivation '/nix/store/583iisxsib0iidl1d1hxg0y8w9f6hqsa-ruby2.6.4-gitaly-proto-1.36.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/msiidn02rbfy5f8fv6bgqqvl6wilyzn2-ruby2.6.4-gitlab-labkit-0.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/pwagbxv5kviqalzw5dchpnr5ws0n999z-gitaly-env.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/lsmq66bqdqaq21ngbkzzvf7k36br2cak-wrapped-ruby-gitaly-env.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1xgv6pnlq69wxw5n1cn5rhyriq004db3-gitaly-1.53.2.drv': 1 dependencies couldn't be built
builder for '/nix/store/kp1hcf93jg8vjy5lycxcanlwlmlfmq6c-wireguard-go-0.0.20190517.drv' failed with exit code 1; last 10 log lines:
  building
  go/src/git.zx2c4.com/wireguard-go/main.go:18:2: cannot find package "golang.zx2c4.com/wireguard/device" in any of:
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/golang.zx2c4.com/wireguard/device (from $GOROOT)
  	/private/tmp/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/device (from $GOPATH)
  go/src/git.zx2c4.com/wireguard-go/main.go:19:2: cannot find package "golang.zx2c4.com/wireguard/ipc" in any of:
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/golang.zx2c4.com/wireguard/ipc (from $GOROOT)
  	/private/tmp/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/ipc (from $GOPATH)
  go/src/git.zx2c4.com/wireguard-go/main.go:20:2: cannot find package "golang.zx2c4.com/wireguard/tun" in any of:
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/golang.zx2c4.com/wireguard/tun (from $GOROOT)
  	/private/tmp/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/tun (from $GOPATH)
cannot build derivation '/nix/store/qmjxj6iqzwjlqqwanfzc08c1qpqxxgvl-wireguard-tools-0.0.20190913.drv': 1 dependencies couldn't be built
hash mismatch in fixed-output derivation '/nix/store/v8x2i02hkj88zylwhkjpfqskbbxlv8p6-conftest-0.12.0-go-modules':
  wanted: sha256:023j34yjgp3asysrjbls86nvhf51d1n86ll3db493rsivlcm0m5q
  got:    sha256:0p91rw8vzp3d7b73cdfni1v5mw75zzcf0yk5f5bcznxc3j3r4ild
cannot build derivation '/nix/store/1abngpgh9za76j9x9spn2qp04sgjlg41-conftest-0.12.0.drv': 1 dependencies couldn't be built
builder for '/nix/store/9g2dvqiy244z8lhg4ssazkk8hawwg2x9-prometheus-2.12.0.drv' failed with exit code 1; last 10 log lines:
  github.com/prometheus/prometheus/tsdb/cmd/tsdb
  github.com/prometheus/prometheus/tsdb/test
  go build github.com/prometheus/prometheus/tsdb/test: no non-test Go files in /private/tmp/nix-build-prometheus-2.12.0.drv-0/go/src/github.com/prometheus/prometheus/tsdb/test
  github.com/prometheus/prometheus/tsdb/testutil
  github.com/prometheus/prometheus/tsdb/tsdbutil
  running tests
  compilation error :go: cloud.google.com/go@v0.44.1: Get https://proxy.golang.org/cloud.google.com/go/@v/v0.44.1.mod: dial tcp: lookup proxy.golang.org: no such host
   
  FAIL	github.com/prometheus/prometheus/cmd/prometheus	24.188s
  FAIL
builder for '/nix/store/1r8ywf0d2xmxf922zysp8dqf7dm5mb26-hyperledger-fabric-1.3.0.drv' failed with exit code 1; last 10 log lines:
          No future change is possible.  Bailing out early after 0.485s.
          Expected
              <int>: 1
          to match exit code:
              <int>: 2
      --- FAIL: TestPluginLoadingFailure/ENDORSERS_ESCC (0.50s)
          testing.go:864: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
  FAIL
  FAIL	github.com/hyperledger/fabric/peer	41.888s
  FAIL
builder for '/nix/store/zncvy20r3vls5c14dqr39aql77wqy9ws-wal-g-0.2.0.drv' failed with exit code 1; last 10 log lines:
  io/ioutil.ReadAll(...)
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/io/ioutil/ioutil.go:45
  github.com/wal-g/wal-g/walg_test_test.TestEncryptionCycle(0xc000126100)
  	/private/tmp/nix-build-wal-g-0.2.0.drv-0/go/src/github.com/wal-g/wal-g/walg_test/crypto_test.go:84 +0x2ec
  testing.tRunner(0xc000126100, 0x47eb8d8)
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:909 +0xc9
  created by testing.(*T).Run
  	/nix/store/8pj3xl5xjh5gff89c2mx725z51smg049-go-1.13/share/go/src/testing/testing.go:960 +0x350
  FAIL	github.com/wal-g/wal-g/walg_test	2.569s
  FAIL
builder for '/nix/store/7a3bqkwfairlgnixvh6ic966rbk4vyw6-openshift-origin-3.11.0.drv' failed with exit code 2; last 10 log lines:
  configuring
  building
  hack/build-go.sh cmd/oc cmd/openshift 
  ++ Building go targets for darwin/amd64: cmd/oc cmd/openshift
  which: no gdu in (/private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go/src/github.com/openshift/origin/_output/local/bin/darwin/amd64:/private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go:/nix/store/50d4z6i73rkb9wc7qh1083csxsxws7gk-go-1.12.9/share/go:/nix/store/50d4z6i73rkb9wc7qh1083csxsxws7gk-go-1.12.9/share/go/bin:/nix/store/2zjv3km94yfl577qnhgxydrzmzn14qpi-remove-references-to/bin:/nix/store/50d4z6i73rkb9wc7qh1083csxsxws7gk-go-1.12.9/bin:/nix/store/w1ndj6ah2y7gbxjax7hlwrd2k319i850-govers-20160623-77fd787-bin/bin:/nix/store/hi40vi42w2rnf9pflyhmsalcj5jf2bsv-clang-wrapper-7.1.0/bin:/nix/store/m6jwkbhyjn5yip6fblh4g45sah01y2dn-clang-7.1.0/bin:/nix/store/x6a3r9rsazlildaxzqay73scy6nv1inz-coreutils-8.31/bin:/nix/store/mybndcc9i2k8bbjh3i8g26g8c8ghjmf1-cctools-binutils-darwin-wrapper/bin:/nix/store/akr0429h46w5v1vd8wh6m848y2r5m0zw-cctools-binutils-darwin/bin:/nix/store/x6a3r9rsazlildaxzqay73scy6nv1inz-coreutils-8.31/bin:/nix/store/bp1vc6ar57f5jby6sgfy0lr7iyqfyk4w-which-2.21/bin:/nix/store/1k91cpc6wbvzvi9dagx26l2h4js02gb9-rsync-3.1.3/bin:/nix/store/vpw7sfgi97757dkx4h0wxfgzp8zxvzl6-go-bindata-20151023-a0ff256-bin/bin:/nix/store/q1f1pmchkkx8yh9g0wz7gm68n2g0qq6h-libkrb5-1.17-dev/bin:/nix/store/hy8x74slxcmzihwj5sx9qngwl2scs34z-libkrb5-1.17/bin:/nix/store/hi40vi42w2rnf9pflyhmsalcj5jf2bsv-clang-wrapper-7.1.0/bin:/nix/store/m6jwkbhyjn5yip6fblh4g45sah01y2dn-clang-7.1.0/bin:/nix/store/x6a3r9rsazlildaxzqay73scy6nv1inz-coreutils-8.31/bin:/nix/store/mybndcc9i2k8bbjh3i8g26g8c8ghjmf1-cctools-binutils-darwin-wrapper/bin:/nix/store/akr0429h46w5v1vd8wh6m848y2r5m0zw-cctools-binutils-darwin/bin:/nix/store/x6a3r9rsazlildaxzqay73scy6nv1inz-coreutils-8.31/bin:/nix/store/x6a3r9rsazlildaxzqay73scy6nv1inz-coreutils-8.31/bin:/nix/store/n28lrd2d4r8igfbwy5g3r6661g0jy3ry-findutils-4.7.0/bin:/nix/store/318c43wf18r10zaykachddkcymm7br5q-diffutils-3.7/bin:/nix/store/bzyz51whcvcpy2xpdc2hf6yrrvjyawdj-gnused-4.7/bin:/nix/store/4srqly43mp9b7jnb3cxiz92la2s2z5rs-gnugrep-3.3/bin:/nix/store/ykv2r00m968fvxhkcv2n6hm37qilz9m3-gawk-5.0.1/bin:/nix/store/qmmjygjchy30pkkcm56hiy6amp1nlafy-gnutar-1.32/bin:/nix/store/7xyf9300r9rpliwksmmghm49rgj79c6g-gzip-1.10/bin:/nix/store/7z18kllrbd44vdy6ri93nv15gj09bcxd-bzip2-1.0.6.0.1-bin/bin:/nix/store/zjpd32jlgl0rhy08k19mr33sshwf02g0-gnumake-4.2.1/bin:/nix/store/9jd5ihjm10rwm3m6aj1wwbyx7f3cc902-bash-4.4-p23/bin:/nix/store/ma17dsn2d1rw3wjldmmn1i6mas64r4rn-patch-2.7.6/bin:/nix/store/sdw0bhkj2jyk9hkrng4s2fx47v9f4w6l-xz-5.2.4-bin/bin)
  /private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go/src/github.com/openshift/origin/hack/lib/util/text.sh: line 49: tput: command not found
  /private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go/src/github.com/openshift/origin/hack/lib/util/text.sh: line 25: tput: command not found
  /private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go/src/github.com/openshift/origin/hack/lib/util/text.sh: line 25: tput: command not found
  /private/tmp/nix-build-openshift-origin-3.11.0.drv-0/go/src/github.com/openshift/origin/hack/lib/util/text.sh: line 25: tput: command not found
  make: *** [Makefile:43: all] Error 127
builder for '/nix/store/lfgnrkbk8mpmywd40ylbr9mgb97vzrhn-qtwebengine-5.11.3.drv' failed with exit code 2; last 10 log lines:
  warning: unknown warning option '-Wno-class-memaccess'; did you mean '-Wno-class-varargs'? [-Wunknown-warning-option]
  1 warning generated.
  ninja: build stopped: subcommand failed.
  Command '['ninja', '-C', '/private/tmp/nix-build-qtwebengine-5.11.3.drv-0/qtwebengine-everywhere-src-5.11.3/src/3rdparty/chromium/tools/gn/out/Release', '-w', 'dupbuild=err', 'gn']' returned non-zero exit status 1
  Project ERROR: GN build error!
  make[2]: *** [Makefile:60: sub-gn-pro-make_first] Error 3
  make[2]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.11.3.drv-0/qtwebengine-everywhere-src-5.11.3/src/buildtools'
  make[1]: *** [Makefile:54: sub-buildtools-make_first] Error 2
  make[1]: Leaving directory '/private/tmp/nix-build-qtwebengine-5.11.3.drv-0/qtwebengine-everywhere-src-5.11.3/src'
  make: *** [Makefile:47: sub-src-make_first] Error 2
cannot build derivation '/nix/store/70gqrqwhgs8qc9c2idyiaxrxp2bngq03-qsyncthingtray-0.5.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g80113lh9823qjqc47i3v0akdkw0sdlc-env.drv': 17 dependencies couldn't be built
[0 built (16 failed), 0.0 MiB DL]
error: build of '/nix/store/g80113lh9823qjqc47i3v0akdkw0sdlc-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/68135
21 package are marked as broken and were skipped:
aws-okta bazel-remote bettercap buildah datadog-agent interlock journalbeat journalbeat6 journalbeat7 magnetico meguca pond prometheus-postfix-exporter prometheus-snmp-exporter python27Packages.syncthing-gtk python37Packages.syncthing-gtk sampler syncthing-gtk syncthing-tray systrayhelper terragrunt

17 package failed to build:
circleci-cli conftest dcrd dcrwallet git-lfs gitaly hyperledger-fabric openshift packetbeat prometheus prometheus-apcupsd-exporter prometheus-blackbox-exporter prometheus-varnish-exporter qsyncthingtray wal-g wireguard-go wireguard-tools

557 package were build:
aerc amass amazon-ecr-credential-helper aminal antibody aptly archiver argo asmfmt assh autospotting awless aws-env aws-iam-authenticator aws-rotate-key aws-vault azure-storage-azcopy azure-vhd-utils buildozer bazelisk blockbook blsd boohu boringssl bosun brig browserpass browsh buildkit buildkite-agent buildkite-agent3 c14 caddy cayley certigo certstrap cfssl chezmoi clair claws cli53 cloud-nuke cloud-print-connector cloud-sql-proxy cloudflared cloudfoundry-cli cockroachdb codesearch cointop compile-daemon confd consul consul-alerts consul-template container-linux-config-transpiler coredns corgi cri-tools croc ctop cue curl_unix_socket dashing datadog-process-agent dbmate deadcode deis deisctl delve demoit dep dep2nix desync devd dex-oidc dgraph direnv diskrsync dive dnscrypt-proxy2 docker docker-credential-gcr docker-distribution docker-gc docker-ls docker-machine docker-machine-kvm docker-machine-kvm2 docker-machine-xhyve docker_18_09 doctl docui documize-community drive drone drone-cli duplicacy easyjson echoip ecs-agent ejson eksctl elfinfo elvish envconsul envdir envsubst errcheck evmdis exercism fac filebeat filebeat7 filegive fluxctl fly fsql fzf gauge gawp gcsfuse gdrive gh-ost ghq git-lfs1 git-sizer git-town gitAndTools.git-annex-remote-b2 gitAndTools.git-appraise gitAndTools.git-bug gitAndTools.grv gitAndTools.hub gitAndTools.lab lefthook gitea gitlab-runner gitlab-shell gitlab-workhorse glide gllvm glock gnatsd go go-2fa go-bindata go-bindata-assetfs go-check go-dependency-manager go-ethereum go-jira go-jsonnet go-junit-report go-langserver go-md2man go-motion go-mtpfs go-outline go-protobuf go-repo-root go-symbols go-tools go-upower-notify go2nix goa gocode gocode-gomod goconst goconvey gocryptfs gocyclo godef gofumpt gogetdoc gogs gohai goklp golangci-lint golint gometalinter gomodifytags gomuks gopass gopkgs goreman gosec gotags gotests gotools gotop gotty govc govendor govers gox grafana_reporter gron grpcurl gsctl guardian-agent gucci gx gx-go harmonist hcloud heapster heartbeat heartbeat7 helmfile heptio-ark hetzner-kube hivemind holochain-go hologram hound httplab hugo hydron i3cat iamy ical2org iferr impl ineffassign influxdb interfacer ipfs-cluster ipfs-migrator ipget jd jid jmespath joker jp json2hcl jump jx k6 k9s kail kbfs kexpand keybase kind kompose kontemplate kops kops_1_12 kt kube-aws kube-prompt kubecfg kubectl kubectx kubeless kubernetes kubernetes-helm kubeval kubicorn kustomize lazydocker lazygit leaps lego lf linkerd linuxkit loccount madonctl mage mailhog maligned manul marathonctl massren matterbridge matterircd mattermost mautrix-whatsapp meme mesos-dns metricbeat metricbeat7 micro miniflux minify minikube minio minio-client mirrorbits mkcert mm mmake mmark moby mod modd mongodb-tools mop morph morty mqtt-bench mtail mustache-go mynewt-newt nats-streaming-server nfpm nginx-sso ngrok-1 nix-prefetch-docker nix-store-gcs-proxy nixops-dns nomad notary noti nsq oauth2_proxy obfs4 oci-image-tool oh out-of-tree overmind pachyderm packer packet packetbeat7 pcstat pdfcpu peco perkeep pet pgcenter pgmetrics phraseapp-client pigeon platinum-searcher powerline-go pprof prom2json prometheus-alertmanager prometheus-aws-s3-exporter prometheus-bind-exporter prometheus-collectd-exporter prometheus-consul-exporter prometheus-dnsmasq-exporter prometheus-dovecot-exporter prometheus-fritzbox-exporter prometheus-gitlab-ci-pipelines-exporter prometheus-haproxy-exporter prometheus-json-exporter prometheus-mesos-exporter prometheus-minio-exporter prometheus-mysqld-exporter prometheus-nginx-exporter prometheus-node-exporter prometheus-openvpn-exporter prometheus-postgres-exporter prometheus-pushgateway prometheus-rabbitmq-exporter prometheus-statsd-exporter prometheus-surfboard-exporter prometheus-unifi-exporter proto-contrib protoc-gen-doc prototool pup python37Packages.jupyter-repo2docker qbec qr-filetransfer quicktemplate rclone reflex reftools restic restic-rest-server reviewdog richgo s3gof3r saml2aws scaleway-cli sensu-go-agent sensu-go-backend sensu-go-cli serfdom shadowfox shfmt sift simplehttp2server skaffold skopeo skydns slack-term smimesign sonobuoy sops ssm-agent statik stern styx subfinder sudolikeaboss syncthing syncthing-cli syncthing-discovery syncthing-relay systemd-journal2gelf tegola teleconsole telegraf teleport terminal-parrot terracognita terraform terraform-docs terraform-inventory terraform-providers.acme terraform-providers.alicloud terraform-providers.ansible terraform-providers.archive terraform-providers.arukas terraform-providers.atlas terraform-providers.aws terraform-providers.azuread terraform-providers.azurerm terraform-providers.azurestack terraform-providers.bigip terraform-providers.bitbucket terraform-providers.brightbox terraform-providers.chef terraform-providers.circonus terraform-providers.clc terraform-providers.cloudflare terraform-providers.cloudscale terraform-providers.cloudstack terraform-providers.cobbler terraform-providers.consul terraform-providers.datadog terraform-providers.digitalocean terraform-providers.dme terraform-providers.dns terraform-providers.dnsimple terraform-providers.docker terraform-providers.dyn terraform-providers.external terraform-providers.fastly terraform-providers.flexibleengine terraform-providers.gandi terraform-providers.github terraform-providers.gitlab terraform-providers.google terraform-providers.google-beta terraform-providers.grafana terraform-providers.hcloud terraform-providers.hedvig terraform-providers.helm terraform-providers.heroku terraform-providers.http terraform-providers.huaweicloud terraform-providers.ibm terraform-providers.icinga2 terraform-providers.ignition terraform-providers.influxdb terraform-providers.kubernetes terraform-providers.librato terraform-providers.linode terraform-providers.local terraform-providers.logentries terraform-providers.logicmonitor terraform-providers.mailgun terraform-providers.matchbox terraform-providers.mysql terraform-providers.netlify terraform-providers.newrelic terraform-providers.nixos terraform-providers.nomad terraform-providers.ns1 terraform-providers.nsxt terraform-providers.null terraform-providers.nutanix terraform-providers.oci terraform-providers.oneandone terraform-providers.opc terraform-providers.openstack terraform-providers.opentelekomcloud terraform-providers.opsgenie terraform-providers.oraclepaas terraform-providers.ovh terraform-providers.packet terraform-providers.pagerduty terraform-providers.panos terraform-providers.postgresql terraform-providers.powerdns terraform-providers.profitbricks terraform-providers.rabbitmq terraform-providers.rancher terraform-providers.random terraform-providers.rightscale terraform-providers.rundeck terraform-providers.runscope terraform-providers.scaleway terraform-providers.secret terraform-providers.segment terraform-providers.selectel terraform-providers.skytap terraform-providers.softlayer terraform-providers.spotinst terraform-providers.statuscake terraform-providers.telefonicaopencloud terraform-providers.template terraform-providers.tencentcloud terraform-providers.terraform terraform-providers.tfe terraform-providers.tls terraform-providers.triton terraform-providers.ucloud terraform-providers.ultradns terraform-providers.vault terraform-providers.vcd terraform-providers.vsphere terraform-providers.yandex terraform_0_11 tewisay textql tflint thanos tilt timescaledb-parallel-copy timescaledb-tune todoist todolist toxiproxy traefik trezord tychus ua uchiwa unconvert up vale vault vgo2nix vimPlugins.youcompleteme vndr vultr webhook websocketd wego ws wuzz xmloscopy xmonad-log xmpp-client xurls yaml2json ycmd yj yq-go zabbixctl

[0.0 MiB DL]
error: build log of '/nix/store/1abngpgh9za76j9x9spn2qp04sgjlg41-conftest-0.12.0.drv' is not available
[0.0 MiB DL]
error: build log of '/nix/store/1xgv6pnlq69wxw5n1cn5rhyriq004db3-gitaly-1.53.2.drv' is not available
[0.0 MiB DL]
error: build log of '/nix/store/70gqrqwhgs8qc9c2idyiaxrxp2bngq03-qsyncthingtray-0.5.8.drv' is not available
[0.0 MiB DL]
error: build log of '/nix/store/qmjxj6iqzwjlqqwanfzc08c1qpqxxgvl-wireguard-tools-0.0.20190913.drv' is not available
```
</details>

c00w commented:
Can I get you to cherry pick https://github.com/c00w/nixpkgs/commit/ce36a3ecc25393080142f10a973e4c86937f8d5d and 
https://github.com/c00w/nixpkgs/commit/b26d7dbdbca5f3a701059ff540b30185b4388285

That should fix prometheus and conftest via downgrade.

c00w committed:
all-packages: use go 1.12 for circleci-cli

c00w committed:
all-packages: use go 1.12 for git-lfs

c00w commented:
@rvolosatovs Any chance we can get the rest of the downgrades from my branch cherry picked, and then the WIP dropped? I think we can probably merge this as is (unless the darwin failures are blocking).

marsam commented:
@GrahamcOfBorg build openshift

marsam reviewed (approved):
I've reviewed the failing packages on Darwin
<details>

|  package  | build status |
|------------------------------|------------------ |
| circleci-cli                 | ✓                |
| conftest                     | ✓                |
| git-lfs                      | ✓                |
| prometheus-varnish-exporter  | ✓                |
| prometheus                   | ✓                |
| dcrd                         | ✓                |
| dcrwallet                    | ✓                |
| gitaly                       | broken in master |
| hyperledger-fabric           | ✓                |
| openshift                    | ✓                |
| packetbeat                   | broken in master |
| prometheus-apcupsd-exporter  | ✓                |
| prometheus-blackbox-exporter | ✓                |
| qsyncthingtray               | broken in master |
| wireguard-go                 | broken in master |
| wireguard-tools              | broken in master |
</details>

kalbasit commented:
> It seems like Go records the lookup/download time? Maybe we can mitigate that by making Go think we're at Unix epoch in `buildGoModule`?

@rvolosatovs is this still relevant? Was it mitigated/fixed?

d-goldin commented:
Do we know if this release also fixes the problematic `TestGCSys` test? https://github.com/golang/go/commit/aae0b5b0b26bf4fd26cad0111535d703691a9083 supposedly should have fixed it, but when trying it on top of 1.12.9 it still didn't seem to work. Maybe someone could give it a run on ofBorg for i686 and aarch64 (though supposedly people observed this on x86_64 too, but I have not seen such failures on our infras or locally)
 
Also see https://github.com/NixOS/nixpkgs/pull/69359

ngerstle-cognite commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7911,7 +7936,11 @@ in
+    inherit (darwin.apple_sdk.frameworks) Security Foundation;
+  };
+
+  go = go_1_13;
```
It would be very nice to break this PR into two- one that introduces the go-1.13 version, and another (or several) to handle the migration of other packages.
Or if this PR is close to done, as indicated by @c00w, let's finish it up..

rvolosatovs commented:
Sorry that it's taking so long, I've been sick for the last couple of days. I will try to get back to this(and hopefully make it ready to merge) today

c00w commented:
No worries :D glad to hear you're feeling better. Please let me know if
there is anything I can do to help.

On Thu, Oct 3, 2019, 02:45 Roman Volosatovs <notifications@github.com>
wrote:

> Sorry that it's taking so long, I've been sick for the last couple of
> days. I will try to get back to this(and hopefully make it ready to merge)
> today
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/68135?email_source=notifications&email_token=AADWWN6ZEO5KJ5IZDOU3LDTQMWIJDA5CNFSM4IT4OFE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAHFPDA#issuecomment-537810828>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADWWN6NNQJLZMWERL2L2SDQMWIJDANCNFSM4IT4OFEQ>
> .
>

rvolosatovs committed:
go_1_13: add kalbasit to maintainers

rvolosatovs commented:
@c00w I rebased this on top of latest master, made sure commits are consistent and removed `[WIP]` label.

c00w commented:
LGTM - any way we can kick the bot to do a full rebuild?

kalbasit commented:
> It seems like Go records the lookup/download time? Maybe we can mitigate that by making Go think we're at Unix epoch in `buildGoModule`?

@rvolosatovs is this still relevant? Was it mitigated/fixed?

c00w commented:
@kalbasit Apologies for jumping in here, but the issue was that conftest was not deterministically fetching it's modules. You can see form the diff calculated above that it appears to be a go sumdb issue. Specifically a new inclusion proof was generated each time the modules were fetched and resolved. This is a new feature introduced in go 1.13. We had a mitigation of leaving conftest at go1.12.

However you got me looking at this a little more closely, and while I need to file a bug with the go team (since you shouldn't get a different inclusion proof each time the module is downloaded), the following commit upgrades conftest to go1.13 and merely disables the sum database. https://github.com/c00w/nixpkgs/commit/2b2778ebc513375ff80770f8509544277a78ccbc @rvolosatovs Any chance I could get you to cherry pick that commit as well?

I think that with that commit (or with leaving conftest at go1.12) this code can be merged safely. This appears to be a super rare corner case (only 1 package), and I think it's not a bug in nix or go+nix, but rather somewhere in the go sum infrastructure and it's interaction with the go binary.

kalbasit commented:
@c00w Got it, thank you for bringing me up to speed.

If one of the dependencies was re-tagged upstream, it could lead to a mismatch between what's cached and what's coming direct. Not everyone treats tags as immutable, I guess Git needs to force it to be this way! See discussion here https://github.com/kubernetes/test-infra/pull/13918#issuecomment-522624183.

c00w commented:
I've encountered the miss tagging, this is something else. With miss
tagging you normally get a sum error claiming it has been tampered with.
Here you appear to get a unique hash every pull, caused by a changing sumdb
proof (which shows up in the on disk files). The underlying dependencies
don't seem to be changing.

On Sun, Oct 6, 2019, 02:15 Wael Nasreddine <notifications@github.com> wrote:

> @c00w <https://github.com/c00w> Got it, thank you for bringing me up to
> speed.
>
> If one of the dependencies was re-tagged upstream, it could lead to a
> mismatch between what's cached and what's coming direct. Not everyone
> treats tags as immutable, I guess Git needs to force it to be this way! See
> discussion here kubernetes/test-infra#13918 (comment)
> <https://github.com/kubernetes/test-infra/pull/13918#issuecomment-522624183>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/68135?email_source=notifications&email_token=AADWWN4HZJI2Z5QW6V4FIG3QNF66XA5CNFSM4IT4OFE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAOCMOI#issuecomment-538715705>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADWWN3HM6C5NSLIL7Y3SZDQNF66XANCNFSM4IT4OFEQ>
> .
>

rvolosatovs commented:
@c00w cherry-picked and rebased

sauyon commented on pkgs/development/tools/conftest/default.nix:
```diff
@@ -2,27 +2,32 @@
+
+  // Something subtle in the go sum db is causing every download to
+  // get a new sum (and thus breaking the hash). This disables the
+  // fetching of the sum from the go sum database.
```
This seems to be breaking the build; I assume these are meant to be comments :P

rvolosatovs replied:
@c00w fixed-up this issue and some formatting in your commit. See https://github.com/NixOS/nixpkgs/pull/68135/commits/1e2ee2a9554e4fd545366c8e2ae29393dd20027d

c00w commented:
Sgtm - should have tested it after I added the comment, not before.

On Mon, Oct 7, 2019, 17:32 Roman Volosatovs <notifications@github.com>
wrote:

> *@rvolosatovs* commented on this pull request.
> ------------------------------
>
> In pkgs/development/tools/conftest/default.nix
> <https://github.com/NixOS/nixpkgs/pull/68135#discussion_r332293442>:
>
> > +  // Something subtle in the go sum db is causing every download to
> +  // get a new sum (and thus breaking the hash). This disables the
> +  // fetching of the sum from the go sum database.
>
> @c00w <https://github.com/c00w> fixed-up this issue and some formatting
> in your commit. See 1e2ee2a
> <https://github.com/NixOS/nixpkgs/commit/1e2ee2a9554e4fd545366c8e2ae29393dd20027d>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/68135?email_source=notifications&email_token=AADWWN4QAZA2B4Q6FSMU6XLQNPIJZA5CNFSM4IT4OFE2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHFHMQI#discussion_r332293442>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADWWN47LNEY2B3VHQDNYATQNPIJZANCNFSM4IT4OFEQ>
> .
>

shazow commented:
How's this looking? Is there anything I can help with to get this merged? :)

clrprod commented:
AFAIK we're just waiting on review.

kalbasit commented:
nix-review is reporting `gcsfuse` is failing to build. It's still going, I might report more.

```
  unpacking sources
  unpacking source archive /nix/store/rpjcp1vyrqs3h700r0l2gqal22jh5iv3-source
  source root is source
  patching sources
  configuring
  building
  Building subPackage .
  go: cannot find main module, but found vendor/vendor.json in /build/source
        to create a module there, run:
        go mod init
```

kalbasit reviewed (approved):
Approving on the condition of getting `gcsfuse` fixed.

kalbasit commented:
@GrahamcOfBorg build gcsfuse

c00w committed:
gcsfuse: add go.mod and go.sum files.

It looks like go1.12 silently created the module, but go1.13 does not.

c00w commented:
Commit
https://github.com/c00w/nixpkgs/commit/f49b872156ff0af5284dd21463c9ae9a9eecb573

Fixes gcsfuse.

kalbasit commented:
@c00w thanks this has worked
@rvolosatovs I tried pushing @c00w 's commit, but I got permission denied, have you unchecked `Allow edits from maintainers`? Anyway, please pull his commit so I can merge this PR.

rvolosatovs commented:
Cherry-picked https://github.com/c00w/nixpkgs/commit/f49b872156ff0af5284dd21463c9ae9a9eecb573 and rebased on top of latest `master`. Should be good to go.

@kalbasit yes, didn't want things like https://github.com/NixOS/nixpkgs/pull/68135#discussion_r329746828 happen again

kalbasit commented:
@GrahamcOfBorg build gcsfuse jx pet
